config/main.py

```diff
@@ -1770,21 +1771,28 @@ def startup(ctx, interface_name):
         if interface_name is None:
             ctx.fail("'interface_name' is None!")
 
-    if interface_name_is_valid(interface_name) is False:
-        ctx.fail("Interface name is invalid. Please enter a valid interface name!!")
+    intf_fs = parse_interface_in_filter(interface_name)
+    if len(intf_fs) == 1 and interface_name_is_valid(interface_name) is False:
+         ctx.fail("Interface name is invalid. Please enter a valid interface name!!")
 
     log_info("'interface startup {}' executing...".format(interface_name))
+    port_dict = config_db.get_table('PORT')
+    for port_name in port_dict.keys():
+        if port_name in intf_fs:
+            config_db.mod_entry("PORT", port_name, {"admin_status": "up"})
+
+    portchannel_list = config_db.get_table("PORTCHANNEL")
+    for po_name in portchannel_list.keys():
+        if po_name in intf_fs:
+            config_db.mod_entry("PORTCHANNEL", po_name, {"admin_status": "up"})
+
+    subport_list = config_db.get_table("VLAN_SUB_INTERFACE")
+    for sp_name in subport_list.keys():
+        if ps_name in intf_fs:
+            config_db.mod_entry("VLAN_SUB_INTERFACE", po_name, {"admin_status": "up"})
```

Comment:
po_name -> sp_name

Reply:
fixed

Reply:
Still see it as `po_name`

Reply:
can you please provide an output of command with subinterface?

Reply:
this interface range does not support subinterface. You can see for subinterface it will take as individual subinterface name as input parameter.